packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -275,7 +275,7 @@ export default class WebpackConfigGenerator {
         );
       }
     }
-    return webpackMerge(baseConfig, rendererConfig || {}, { entry, output, plugins });
+    return webpackMerge({ entry, output, plugins }, rendererConfig || {}, baseConfig);
```

Comment:
Webpack Merge prioritizes right -> left:

we determine the config target within the baseConfig (i.e 'web', 'electron-preload') and not the one generated from `rendererConfig = this.pluginConfig.renderer.config`.

Reply:
edit: maybe I was wrong and shouldn't have touched this

Reply:
https://yarnpkg.com/package?name=webpack-merge#:~:text=webpack%2Dmerge%20provides%20a%20merge,values%20within%20a%20function%20again.

within the first code example, a comment mentions that keys to the right take precedence. 
I think these need to be moved because otherwise, the separate targets that this PR evaluates do not get translated into the final webpack config (ex. prior to this change, `target: electron-preload` target in the baseconfig gets overwritten  by `target: web` in the rendererConfig, even if its a preload-only entrypoint

I'm gonna dig a bit deeper and test whether I can make a functioning webpack app with these changes